adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala

```diff
@@ -339,6 +339,373 @@ private[adam] class VariantContextConverter extends Serializable with Logging {
+    val filtersPassed = Option(v.getFiltersPassed).getOrElse(false)
+
+    (filtersApplied, filtersPassed) match {
+      case (false, false) => vcb.unfiltered
+      case (false, true)  => vcb.passFilters // log warning?
+      case (true, false)  => vcb.filters(new java.util.HashSet(v.getFiltersFailed()))
+      case (true, true)   => vcb.passFilters
+    }
+    vcb
+  }
+
+  private[converters] def extractSomatic(
+    v: Variant,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    val somatic: java.lang.Boolean = Option(v.getSomatic).getOrElse(false)
+    if (somatic) {
+      vcb.attribute("SOMATIC", true)
+    }
+    vcb
+  }
+
+  private val variantExtractionFns: Iterable[(Variant, VariantContextBuilder) => VariantContextBuilder] = Iterable(
+    extractNames(_, _),
+    extractFilters(_, _),
+    extractSomatic(_, _)
+  )
+
+  // variant annotation conversion functions
+
+  private[converters] def formatAncestralAllele(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttributeAsString("AA", null)).foreach(vab.setAncestralAllele(_))
+    vab
+  }
+
+  private[converters] def formatDbSnp(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttribute("DB").asInstanceOf[java.lang.Boolean]).foreach(vab.setDbSnp(_))
+    vab
+  }
+
+  private[converters] def formatHapMap2(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttribute("H2").asInstanceOf[java.lang.Boolean]).foreach(vab.setHapMap2(_))
+    vab
+  }
+
+  private[converters] def formatHapMap3(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttribute("H3").asInstanceOf[java.lang.Boolean]).foreach(vab.setHapMap3(_))
+    vab
+  }
+
+  private[converters] def formatValidated(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttribute("VALIDATED").asInstanceOf[java.lang.Boolean]).foreach(vab.setValidated(_))
+    vab
+  }
+
+  private[converters] def formatThousandGenomes(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    Option(vc.getAttribute("1000G").asInstanceOf[java.lang.Boolean]).foreach(vab.setThousandGenomes(_))
+    vab
+  }
+
+  private[converters] def formatAlleleCount(
+    vc: HtsjdkVariantContext,
+    vab: VariantAnnotation.Builder,
+    v: Variant,
+    index: Int): VariantAnnotation.Builder = {
+
+    val ac = vc.getAttributeAsList("AC")
+    if (ac.size > index) {
+      vab.setAlleleCount(ac.get(index).asInstanceOf[java.lang.Integer])
```

Comment:
Yeah, you're probably going to be disappointed. If my experience with HTSJDK's Genotype attributes is mirrored in the HTSJDK Variant world... all of the keys will be stored as strings.